File: Electronics_Shop_17.Services/Database/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Electronics_Shop_17.Model.Requests;

namespace Electronics_Shop_17.Services.Database
{
    public class Order : IHelperId,ISoftDelete
    {
        public int Id { get; set; }
        public DateTime OrderTime { get; set; }
        public double TotalAmount { get; set; }
        public double? FinalTotalAmount { get; set; }
        public string StateMachine { get; set; }

        public int CustomerId { get; set; }
        public virtual Customer Customer { get; set; }
        public int AdressId { get; set; }
        public virtual Adress Adress { get; set; }
        public int? CouponId { get; set; }
        public virtual Coupon Coupon { get; set; }
        public int PaymentMethodId { get; set; }
        public virtual PaymentMethod PaymentMethod { get; set; }
        public virtual List<OrderItem> OrderItems { get; set; }
        public bool isDeleted { get; set; }
        public string? PaymentId { get; set; }
        public string? PaymentIntent { get; set; }
    }
}

File: angular17_web/src/app/models/order/order.ts
import { Adress } from '../adress/adress';
import { Coupon } from '../coupon/coupon';
import { PaymentMethod } from '../payment-method/payment.method';
import { OrderItem } from '../order-item/order.item';

export class Order {
  id!: number;
  orderTime!: Date;
  totalAmount!: number;
  finalTotalAmount?: number;
  stateMachine!: string;
  adress!: Adress;
  couponId?: number;
  coupon?: Coupon;
  paymentMethod!: PaymentMethod;
  orderItems: OrderItem[] = [];

  constructor(data?: any) {
    if (!data) return;

    this.id = data.id;
    this.orderTime = data.orderTime ? new Date(data.orderTime) : undefined as any;
    this.totalAmount = data.totalAmount;
    this.finalTotalAmount = data.finalTotalAmount;
    this.stateMachine = data.stateMachine;
    this.adress = data.adress ? new Adress(data.adress) : undefined as any;
    this.couponId = data.couponId;
    this.coupon = data.coupon ? new Coupon(data.coupon) : undefined;
    this.paymentMethod = data.paymentMethod ? new PaymentMethod(data.paymentMethod) : undefined as any;
    this.orderItems = Array.isArray(data.orderItems)
      ? data.orderItems.map((x: any) => new OrderItem(x))
      : [];
  }
}
